ios/Trifle/Sources/Trifle.swift

```diff
@@ -62,24 +62,54 @@ public class Trifle {
     }
     
     /**
-     // TODO(dcashman): define message format //
      Sign the provided data with the provided key, including appropriate Trifle
      metadata, such as the accompanying certificate.
 
      - parameters: data - raw data to be signed.
      - parameters: keyHandle - key handle used for the signing.
-     - parameters: certificate - certificate to be included in the SignedData message.
+     - parameters: certificate list - certificate chain to be included in the SignedData message.
         Must match the key in keyHandle.
 
      - returns:`SignedData` - signed data message in the Trifle format.
     */
     public func createSignedData(
         data: Data,
         keyHandle: KeyHandle,
-        certificate: Certificate
-    ) -> SignedData {
-        // TODO: IMPLEMENT
-        SignedData()
+        certificate: Array<Certificate>
+    ) throws -> SignedData {
+
+        guard !data.isEmpty && !certificate.isEmpty else {
+            throw TrifleError.invalidInput("Data or Certifcate should not be empty.")
+        }
+        
+        // TODO: check key handle domain matches the one in trifle
+        
+        // TODO: check leaf cert mactches the public key that will be used for signing
+        
+        //check cert chain validates
+        // already checked certificate chain is not empty
+        if (certificate.count == 1 ) {
```

Comment:
if we change up the prototype, we can now do something like

```
        guard leafCert.verify(intermediateChain: Array(certificate.dropFirst())) else {
            throw TrifleError.invalidCertificateChain
        }
```

Reply:
I personally like changing the proto better but that also means that if root cert is separate from certChain, we will require app to put them together before calling us.
Im personally in favor of that.

@dcashman what do you think of changing:
`    public func verify(
        certificateRequest: MobileCertificateRequest? = nil,
        intermediateChain: Array<Certificate>,
        rootCertificate: Certificate? = nil )
`
to
`    public func verify(
        certificateRequest: MobileCertificateRequest? = nil,
        intermediateChain: Array<Certificate>  )
`
and requiring caller to append rootCert to certChain?

Reply:
this doesn't work because the itnermediateChain() is what is provided by the CA, whereas the rootCertificate is what is provided by the client.  But another way, verify() needs to know whether or not to trust the cert chain, rather than any valid cert chain.  Appending doesn't necessarily work because it's possible that the root-cert could actually be not the last element of the chain.

Reply:
sorry for the confusion. when I say prototype (C land), I meant the method signature.

Reply:
updated.